core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -574,6 +575,8 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
                 }
             } catch (InterruptedException ex) {
                 throw ex;
+            } catch (CorruptIndexException ex) {
```

Comment:
Correct me if I am wrong here but the recovery path looks as follows:

1. RLM will continue to try uploading these segments.
2. It will fail continuously (until the indexes are recovered)
3. Tiered storage lag metric will increase (we have an open ticket to create that metric I think)
4. Operator will fix the corrupted segments.


Alternatively, can we have an automated recovery path here where we will re-create the indexes if they are corrupted before trying upload again. This automated recovery can be done as part of a separate PR.

Reply:
I will doublecheck it, thanks!

Reply:
Generally your understanding is correct. I guess this behaviour for now is better than copying corrupted segment and have problems with that afterwards. 

Recovery path should be done as a separate PR since this problem since this is not specifically Tiered storage problem. Index corruption can be caused for non TS clusters as well. `

My change is only about avoiding this corruption been uploaded to remote storage